File: src/main/java/com/gorogoro/auth/user/application/port/in/GetUserUseCase.java
package com.gorogoro.auth.user.application.port.in;

import com.gorogoro.auth.user.application.dto.result.UserNicknamesResult;
import com.gorogoro.auth.user.application.dto.result.UserResult;

import java.util.List;

public interface GetUserUseCase {
    UserResult getUser(Long userId);
    UserNicknamesResult getUserNicknames(List<Long> userIds);
}

File: src/main/kotlin/com/gorogoro/auth/user/application/port/in/GetUserUseCase.kt
package com.gorogoro.auth.user.application.port.`in`

import com.gorogoro.auth.user.application.dto.GetUserCommand
import com.gorogoro.auth.user.infra.adapter.`in`.server.response.NicknameResponse
import com.gorogoro.auth.user.infra.adapter.`in`.web.response.FindUserResponse
import com.gorogoro.auth.user.model.User

interface GetUserUseCase{
    fun getUserInfo(command: GetUserCommand): FindUserResponse
    fun getUserNickname(command: GetUserCommand): NicknameResponse
}
